### ISSUE TITLE
Style Settings Plug In Not Working

### ISSUE THREAD

**Issue by roux-rook** [oleeskild/obsidian-digital-garden#323]
Hello! I'm trying to use the compatibility with the style settings plug in to customize my site but every time I hit apply it doesnt reflect my customization that Im using on my local device. I can change the theme no problem and have it reflect that but no customizations will show up. I have tried making sure the plug ins are updated and merged prs and still nothing. Is there a step Im missing at all? Any help is greatly appreciated thanks!

**Comment by roux-rook** [comment]
update to this I looked at my commits on my repo here for the project and every time I've attempted to apply my style settings it just does an empty commit for the favicon. I've even made new test repos in new vaults and still get the same issue.

**Comment by oyes77** [comment]
I have the same issue t.t

**Comment by julesvirallinen** [comment]
Any videos and additional context would help! Did manage to recreate the favicon thing though (maybe) https://github.com/julesvirallinen/testing-garden/commit/8c2fe689572611b6f737969dfc91c7521cb76a26

**Comment by DudeThatsErin** [comment]
Recorded a video of this happening...
![video](https://dudethats.me/video.gif)

Also my repo: https://github.com/DudeThatsErin/App-Seeker/commits/main
Current screenshot as of posting this:
![image](https://github.com/oleeskild/obsidian-digital-garden/assets/2681022/eec29ec4-1425-4435-9020-f587b37d0bd2)

This is so obnoxious. Now I have to manually recreate the callouts and edits I did in Style Settings.

**Comment by DudeThatsErin** [comment]
@julesvirallinen I want to make sure you see what I have provided.

**Comment by MahmadSharaf** [comment]
Here is a video showing:

- That Obsidian theme is set to "AnuPuchin"
- Style settings data
- Digital Garden applies the same theme
- Digital Garden applies style settings
- The GitHub commits for both settings
- The final state of the website showing that the default settings of the theme is applied

![Video](https://drive.google.com/file/d/18EPKi8uWwLHeHyOOPpjXaTCbFCWjcncj/view?usp=sharing)

**Comment by julesvirallinen** [comment]
Can you update the plugin and see what happens? We fixed a few bugs here, but maybe not this specific bug yet. 

To fix a step by step guide how to reproduce a bug is best eg

1. install theme AnuPuchin 
2. set specific style setting 
3. ...

**Comment by MahmadSharaf** [comment]
@julesvirallinen Yes, I updated the plugin before doing the exercise in the video

**Comment by MahmadSharaf** [comment]
@julesvirallinen here is a step-by-step guide on how to reproduce the issue:

1. Install and use “AnuPpuccin” theme.
2. Install and enable “Style Settings” community plugin.
3. Apply theme setting shared below.
    ```json
    {
      "anuppuccin-theme-settings@@anp-active-line": "anp-current-line-border",
      "anuppuccin-theme-settings@@anp-alt-rainbow-style": "anp-simple-rainbow-color-toggle",
      "anuppuccin-theme-settings@@anp-rainbow-folder-bg-opacity": 0.7,
      "anuppuccin-theme-settings@@anp-simple-rainbow-title-toggle": true,
      "anuppuccin-theme-settings@@anp-simple-rainbow-icon-folder-toggle": true,
      "anuppuccin-theme-settings@@anp-simple-rainbow-indentation-toggle": true,
      "anuppuccin-theme-settings@@anp-simple-rainbow-collapse-toggle": true,
      "anuppuccin-theme-settings@@anp-simple-rainbow-icon-toggle": false,
      "anuppuccin-theme-settings@@anp-rainbow-subfolder-color-toggle": true,
      "anuppuccin-theme-settings@@anp-header-color-toggle": true,
      "anuppuccin-theme-settings@@anp-h1-color-custom": "anp-h1-rosewater",
      "anuppuccin-theme-settings@@anp-decoration-toggle": true,
      "anuppuccin-theme-settings@@anp-bold-custom": "anp-bold-maroon",
      "anuppuccin-theme-settings@@anp-callout-select": "anp-callout-sleek",
      "anuppuccin-theme-settings@@callout-radius": 12,
      "anuppuccin-theme-settings@@callout-title-padding": "7px",
      "anuppuccin-theme-settings@@anp-tab-stacked-pane-width": 1,
      "anuppuccin-theme-settings@@anp-file-label-align": "0",
      "anuppuccin-theme-settings@@anp-layout-select": "anp-card-layout",
      "anuppuccin-theme-settings@@anp-card-layout-actions": true,
      "anuppuccin-theme-settings@@anp-card-layout-filebrowser": true,
      "anuppuccin-theme-settings@@anuppuccin-accent-toggle": true,
      "anuppuccin-theme-settings@@anuppuccin-theme-dark": "ctp-macchiato"
    }
    ```
4. Navigate to “Digital Garden” settings and click on “Manage appearance”.
5. Under “Theme Settings”,
   - Choose “AnuPpuccin” for “Theme"
   - And “Base Theme” as “Dark”
6. Click on "Apply Settings" button.
7. Click on "Apply" button, under "Style Settings Plugin".

The current content of .env file is:
```env
NOTE_ICON_FILETREE=false
NOTE_ICON_INTERNAL_LINKS=false
NOTE_ICON_BACK_LINKS=false
STYLE_SETTINGS_CSS=body.css-settings-manager { --anp-rainbow-folder-bg-opacity: 0.7; --callout-radius: 12px; --callout-title-padding: 7px; --anp-tab-stacked-pane-width: 1; --anp-file-label-align: 0; } body.theme-light.css-settings-manager { } body.theme-dark.css-settings-manager { }
THEME=https://raw.githubusercontent.com/AnubisNekhet/AnuPpuccin/HEAD/theme.css
BASE_THEME=dark
dgHomeLink=true
```

**Comment by julesvirallinen** [comment]
Thanks! this will help a lot!

**Comment by MahmadSharaf** [comment]
Thanks to you and to Oleeskild for making this plugin updated and feature-rich

**Comment by julesvirallinen** [comment]
Ok so it seems from your example that I guess the problem is that it's just applying a few of your settings? Seeing that the env settings for styles only has some of your settings. 

The way dg styles work seems to be that the generated css is just copied to that env value. 

I managed to change the link weight, and it updated on the site, but there seems to be other values that aren't 

does anyone know what style-settings does with those values? How are they communicated to the theme?

**Comment by MahmadSharaf** [comment]
No, I don't know CSS, but I tried to tweak things around and deduced the below, which could be absolutely wrong:

1. Theme provides a set of options, for example multiple themes.
   1. Mocha. (Default)
   2. Latte
   3. Macchiato
3. Style settings, inserts the CSS line for the chosen theme into Obsidian CSS and removes/replaces all CSS lines related to the other themes.

For example, Style Settings variable for choosing the Macchiato is `"anuppuccin-theme-settings@@anuppuccin-theme-dark": "ctp-macchiato"`. But this is actually translated into:
1. Look into a Theme with id: `anuppuccin-theme-settings`. Style settings searches and find that it is related to AnuPpuccin Theme.
2. Under this file, search for a settigns with id `anuppuccin-theme-dark`. Stle settings searches the file and find the below:
   ```YAML
    id: anuppuccin-theme-dark
    title: Dark Theme Flavor
    description: Select your preferred dark mode flavor
    type: class-select
    allowEmpty: false
    default: ctp-mocha
    options:
      - 
        label: Frappe
        value: ctp-frappe
      - 
        label: Macchiato
        value: ctp-macchiato
      - 
        label: Mocha
        value: ctp-mocha
      - 
        label: Mocha Old
        value: ctp-mocha-old
      - 
        label: Royal Velvet
        value: ctp-royal-velvet
   ```
3. Choose the one with value `ctp-macchiato`.
   1. Style settings searches the file for a CSS with value `ctp-macchiato`
   2. Inject it into Obsidian CSS
   3. Remove the CSS files for other values, which are `ctp-frappe`, `ctp-mocha`, `ctp-mocha-old`, `ctp-royal-velvet`.

What currently Digital Garden doing is error-prone, because it just reads Style Settings settings not interpretting what these settings are actual do to the theme.

**Comment by dir-rafa** [comment]
This issue is bothering me too since I am using a highly customised version of the Anuppuccin theme and I want my digital garden to have the same appearance as my obsidian vault. Therefore I tried to figure out what is causing the desync between the style settings and I want to share my findings:

First of all the current implementation of the synchronisation with the Style Settings plugin only looks at the styles that are applied on the `#css-settings-manager` style element which gets populated by the plugin. However this is not the only way it applies styles to Obsidian. In fact the Style Settings documentation describes **8 different types** of *style settings* (***heading, class-toggle, class-select, variable-text, variable-number, variable-number-slider, variable-select & variable-color***), two of which **"add classes on the body element"**. Therefore I had a look on the theme settings that do not get applied to the digital garden and indeed those were declared as either one of the two beforementioned types (***class-toggle*** or ***class-select***).
That being said I can't really think of an elegant yet simple solution to this problem - one could simply apply every class the body element in Obsidian has to the body element of the website, but I certainly wouldn't classify this as *elegant*. Another option would be to do a whole bunch of regex magic to read the theme file as one giant string and filter out the associated classes, but this hurts my brain real bad..

On a side note, I noticed that the Anuppuccin theme uses the css selector ".app-container" e.g. when applying color to a heading (".anp-header-color-toggle.anp-h1-red .app-container") - this would never be applied on the digital garden site even if the beforementioned issue would be fixed, since the DOM doesn't have an element with the app-container class. I don't know if this selector is also used in other themes, but an easy solution to this problem would be to simply give the "main" element the ".app-container" class.

Hope all of this is comprehensible
